staging/src/k8s.io/apiserver/pkg/storage/etcd/etcd_helper.go

```diff
@@ -347,7 +347,8 @@ func (h *etcdHelper) GetToList(ctx context.Context, key string, resourceVersion
 	metrics.RecordEtcdRequestLatency("get", getTypeName(listPtr), startTime)
 	if err != nil {
 		if etcdutil.IsEtcdNotFound(err) {
-			return nil
+			etcdErr := err.(etcd.Error)
```

Comment:
maybe be a little more fault tolerant... do a conditional cast, and only call UpdateList if etcdErr.Index != 0?

Reply:
Fair enough - fixed.